src/middlewared/middlewared/main.py

```diff
@@ -113,14 +113,19 @@ def __init__(self, middleware, loop, request, response):
 
     @functools.cached_property
     def origin(self):
-        sock = self.request.transport.get_extra_info("socket")
-        if sock.family == socket.AF_UNIX:
-            peercred = sock.getsockopt(socket.SOL_SOCKET, socket.SO_PEERCRED, struct.calcsize('3i'))
-            pid, uid, gid = struct.unpack('3i', peercred)
-            return UnixSocketOrigin(pid, uid, gid)
-
-        remote_addr, remote_port = get_remote_addr_port(self.request)
-        return TCPIPOrigin(remote_addr, remote_port)
+        try:
+            sock = self.request.transport.get_extra_info("socket")
+            if sock.family == socket.AF_UNIX:
```

Comment:
Why is all the block under `AttributeError` handler and not just the relevant part?